README.md

```diff
@@ -132,6 +170,50 @@ automatically even if the code inside the `do` scope throws an error.)
 Conversely, `load(::Stream)` and `save(::Stream)` should not close the
 input stream.
 
+`loadstreaming` and `savestreaming` use the same query mechanism, but return a
+decoded stream that users can `read` or `write`. You should also implement a
+`close` method on your reader or writer type. Just like with `load` and `save`,
+if the user provided a filename, your `close` method should be responsible for
+closing any streams you opened in order to read or write the file. If you are
+given a `Stream`, your `close` method should only do the clean up for your
+reader or writer type, not close the stream.
+
+```jl
+struct WAVReader
+    io::IO
+    ownstream::Bool
+end
+
+function read(reader::WAVReader, frames::Int)
+    # read and decode audio samples from reader.io
+end
+
+function close(reader::WAVReader)
```

Comment:
Likewise, clarify whether this is `Base.close` or a private `close`. (It's fine to explain all in just one place.)

Also, I notice your test implements `eof`, worth mentioning?

Reply:
whoops, should be `Base.[read|write|close]`. I'll fix. re: `eof`, I'd love it if there were a well-defined and documented stream interface we could link to here. I can try at least adding a list of functions to implement here and maybe eventually migrate it to [the julia docs](https://docs.julialang.org/en/stable/manual/interfaces/#)

Reply:
I went down a bit of a rabbit hole trying to figure out a more formalized stream interface. My notes are [here](https://gist.github.com/ssfrr/73a232a5bc52f89d7bf2173f6917274d), but basically it's not super well-defined at this point. Once that's nailed down we can link to it in these docs.